components/brave_referrals/browser/brave_referrals_service.cc

```diff
@@ -412,196 +145,6 @@ void BraveReferralsService::MaybeDeletePromoCodePref() const {
     pref_service_->ClearPref(kReferralPromoCode);
 }
 
-std::string BraveReferralsService::BuildReferralInitPayload() const {
```

Comment:
`BuildReferralInitPayload` and `BuildReferralFinalizationCheckPayload` should remain. `FetchReferralHeaders` should go. `InitReferral`, `CheckForReferralFinalization`, etc. should remain. Basically, we only want to get rid of functionality that's directly tied to the referral headers.